### PR TITLE
Support for CLI command completions

### PR DESCRIPTION
**Pull request by juhoinkinen**
Adds support for tab-key completions for commands, options and project id and vocabulary id parameters via [Click mechanism](https://click.palletsprojects.com/en/8.1.x/shell-completion/#enabling-completion).

A user needs to enable the support as instructed in README.md, i.e. by sourcing the completion script, either manually for every terminal session or automatically for all new sessions by the sourcing the script in a shell startup file (instructed to use `.bashrc`). 

To produce the completion script the `annif completion` CLI command is added. This seems to be the standard way that many CLI tools use:  [pip](https://pip.pypa.io/en/stable/user_guide/#command-completion), [dvc](https://dvc.org/doc/install/completion?tab=Bash-Debian-Ubuntu), [helm](https://helm.sh/docs/helm/helm_completion_bash/).

Implements #684.

### CODE REVIEW

File: annif/cli.py [hunk 1]
```diff
@@ -551,5 +551,21 @@ def run_hyperopt(project_id, paths, docs_limit, trials, jobs, metric, results_fi
...
+@click.option("--fish", "shell", flag_value="fish")
+def completion(shell):
+    """Generate autocompletion script for the given shell."""
+
+    if shell is None:
+        click.echo("Shell not given, try --bash, --zsh or --fish")
+        return
+
+    script = os.popen(f"_ANNIF_COMPLETE={shell}_source annif").read()
+    click.echo("# Generated by Annif")
```
Comment (osma) [inline annif/cli.py:566]: How about adding the Annif version number to the comment? Later versions may have more commands etc.

File: annif/cli.py [hunk 2]
```diff
@@ -56,7 +56,7 @@ def run_list_projects():
 
 
 @cli.command("show-project")
-@click.argument("project_id")
+@click.argument("project_id", shell_complete=cli_util.complete_project_id)
```
Comment (osma) [inline annif/cli.py:59]: How about defining a cli_util.project_id decorator instead of repeating this many times?

**Comment by juhoinkinen** [comment]
The completion response is not very instantaneous but quite usable. On my laptop `annif --help` and `annif --version` take 0.6-0.7 s, but I think the completion response is a bit faster.

I tried to optimize startup speed by moving some import statements into functions in `cli.py` and elsewhere. This cut `annif --version` time in half but did not affect `annif --help` and seemed not to affect completion response time. :(

**Comment by juhoinkinen** [comment]
> The completion response is not very instantaneous but quite usable. On my laptop `annif --help` and `annif --version` take 0.6-0.7 s, but I think the completion response is a bit faster.
> 
> I tried to optimize startup speed by moving some import statements into functions in `cli.py` and elsewhere. This cut `annif --version` time in half but did not affect `annif --help` and seemed not to affect completion response time. :(

The response time of completions will be improved by #696 to ~0.3 seconds or slightly less.

**Pull request merged**